### PR TITLE
1.10 enablement

### PR DESCRIPTION
**Pull request by dkarrasch**
This is to make use of https://github.com/JuliaLang/julia/pull/50058. While that PR has not been merged, I'm putting this up to see whether this is working properly on older julia versions, and then we check nightly later.

### CODE REVIEW

File: lib/mkl/linalg.jl [hunk 1]
```diff
@@ -254,23 +263,34 @@ end
...
                             (:UpperTriangular, 'U', 'N'),
                             (:UnitUpperTriangular, 'U', 'U'))
     @eval begin
         # Multiplication
-        LinearAlgebra.lmul!(A::$t{T,<:oneStridedVecOrMat},
-                            B::oneStridedVecOrMat{T}) where {T<:onemklFloat} =
-            trmm!('L', $uploc, 'N', $isunitc, one(T), parent(A), B, B)
-        LinearAlgebra.rmul!(A::oneStridedVecOrMat{T},
-                            B::$t{T,<:oneStridedVecOrMat}) where {T<:onemklFloat} =
-            trmm!('R', $uploc, 'N', $isunitc, one(T), parent(B), A, A)
```
Comment (dkarrasch) [inline lib/mkl/linalg.jl:269]: By the signature of `trmm!`, only matrix arguments are allowed, so either it should be relaxed over there or, as is done here, restricted here.

File: lib/mkl/wrappers.jl [hunk 2]
```diff
@@ -1100,7 +1100,7 @@ function trmm(side::Char,
               alpha::Number,
               A::oneStridedMatrix{T},
               B::oneStridedMatrix{T}) where T
-    trmm!(side, uplo, transa, diag, alpha, A, B)
+    trmm!(side, uplo, transa, diag, alpha, A, copy(B))
```
Comment (dkarrasch) [inline lib/mkl/wrappers.jl:1103]: I think this was oversight, and the corresponding test required an according change.

File: lib/mkl/linalg.jl [hunk 3]
```diff
@@ -254,23 +263,34 @@ end
...
+LinearAlgebra.generic_trimatmul!(C::oneStridedMatrix{T}, uploc, isunitc, tfun::Function, A::oneStridedMatrix{T}, B::oneStridedMatrix{T}) where {T<:onemklFloat} =
+    trmm!('L', uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', isunitc, one(T), A, C === B ? C : copyto!(C, B))
+LinearAlgebra.generic_mattrimul!(C::oneStridedMatrix{T}, uploc, isunitc, tfun::Function, A::oneStridedMatrix{T}, B::oneStridedMatrix{T}) where {T<:onemklFloat} =
+    trmm!('R', uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', isunitc, one(T), B, C === A ? C : copyto!(C, A))
+LinearAlgebra.generic_trimatdiv!(C::oneStridedMatrix{T}, uploc, isunitc, tfun::Function, A::oneStridedMatrix{T}, B::oneStridedMatrix{T}) where {T<:onemklFloat} =
+    trsm!('L', uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', isunitc, one(T), A, C === B ? C : copyto!(C, B))
+LinearAlgebra.generic_mattridiv!(C::oneStridedMatrix{T}, uploc, isunitc, tfun::Function, A::oneStridedMatrix{T}, B::oneStridedMatrix{T}) where {T<:onemklFloat} =
+    trsm!('R', uploc, tfun === identity ? 'N' : tfun === transpose ? 'T' : 'C', isunitc, one(T), B, C === A ? C : copyto!(C, A))
```
Comment (dkarrasch) [inline lib/mkl/linalg.jl:274]: This adds `rdiv!`-functionality, which could be included in the "old" branch below.

**Comment by dkarrasch** [comment]
Nightly fails, but I don't understand why. Something with StaticArrays.jl?

**Comment by dkarrasch** [comment]
Perhaps we should add testing on v1.10-something also to see whether things work as intended after the alpha2-release.

**Comment by dkarrasch** [comment]
Looks like StaticArraysCore.jl may need an upate, so that tests pass also on nightly?

**Comment by maleadt** [comment]
This also fails outside of this PR, so let's just mark 1.10 as broken for now.

**Pull request merged**